src/server/services/wms/qgswmsrenderer.cpp

```diff
@@ -2307,6 +2308,19 @@ namespace QgsWms
           }
         }
       }
+      //add maptip attribute based on html/expression
+      QString mapTip = layer->mapTipTemplate();
+      if ( !mapTip.isEmpty() && mWmsParameters.withMapTip() )
+      {
+        QDomElement maptipElem = infoDocument.createElement( QStringLiteral( "Attribute" ) );
+        maptipElem.setAttribute( QStringLiteral( "name" ), QStringLiteral( "maptip" ) );
+        QgsExpressionContext context { renderContext.expressionContext() };
+        QgsExpressionContextScope *scope = QgsExpressionContextUtils::layerScope( layer );
```

Comment:
Here we probably want globalProjectLayerScopes and add an extra scope for the additional variable.

Reply:
done in https://github.com/qgis/QGIS/pull/55680/commits/b0e2e99b658c90546ea305eefdfcbdf5f0e3f419